### PR TITLE
Use namedObject to parse AllocationCommands

### PR DESCRIPTION
**Pull request by nik9000**
This removes `AllocationCommandRegistry` entirely and replaces
it with `XContentParser#namedObject`, removing another class from
guice.

### CODE REVIEW

**Comment by nik9000** [comment]
@javanna, could I trouble you to have a look at this? If someone else wants it, I'm not picky.

File: core/src/main/java/org/elasticsearch/common/network/NetworkModule.java [hunk 1]
```diff
@@ -76,12 +75,8 @@
     private final Settings settings;
     private final boolean transportClient;
 
-    private static final AllocationCommandRegistry allocationCommandRegistry = new AllocationCommandRegistry();
     private static final List<NamedWriteableRegistry.Entry> namedWriteables = new ArrayList<>();
-
-    private final Map<String, Supplier<Transport>> transportFactories = new HashMap<>();
```
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/common/network/NetworkModule.java]: I moved these below that static initializer block.

File: core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java [hunk 2]
```diff
@@ -111,31 +105,14 @@ protected void addCustomFields(XContentBuilder builder, ClusterRerouteResponse r
...
-        private ParseContext(AllocationCommandRegistry registry, ParseFieldMatcher parseFieldMatcher) {
-            this.registry = registry;
-            this.parseFieldMatcher = parseFieldMatcher;
-        }
-
-        @Override
-        public ParseFieldMatcher getParseFieldMatcher() {
-            return parseFieldMatcher;
-        }
-    }
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java]: oh seems like you also removed some ParseFieldMatcher usages, nice :)

**Review by javanna — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java (hunk 2 above)
Comment (nik9000) [inline core/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestClusterRerouteAction.java]: I'm helping!

**Comment by nik9000** [comment]
Thanks for reviewing @javanna!

**Pull request merged**